python/ember/input.py

```diff
@@ -1314,6 +1342,183 @@ def run(self):
         solver.finalize()
         return solver
 
+    def runESR(self):
+        """
+        Run a sequence of flame simulations at increasing strain rates until
+        a steady burning solution is no longer possible with an increase
+        in strain rate. The parameters governing this progression are defined
+        under the configuration field :attr:`Extinction`.
+        """
+        import shutil
+
+        if os.path.exists(self.paths.outputDir):
+            print('Output directory already exists')
+            print('Exiting...')
+            return
+
+        outDirTop = self.paths.outputDir
+        logPath = self.paths.outputDir+'/logFiles'
+        currentRunPath = self.paths.outputDir+'/runFiles'
+        ssProfilePath = self.paths.outputDir+'/ssFiles'
+        restartPath = None
+
+        self.paths.outputDir = currentRunPath
+
+        if not os.path.exists(self.paths.outputDir):
+            os.makedirs(outDirTop, 0o0755)
+            os.makedirs(logPath, 0o0755)
+            os.makedirs(currentRunPath, 0o0755)
+            os.makedirs(ssProfilePath, 0o0755)
+
+        fileExt = self.outputFiles.fileExtension
+
+        if self.paths.logFile:
+            _logFile = open(self.paths.logFile, 'w')
+            def log(message):
+                _logFile.write(message)
+                _logFile.write('\n')
+                _logFile.flush()
+        else:
+            def log(message):
+                print(message)
+
+        strainRateValues = []
+        maxTemps = []
+
+        if self.extinction.method is 'step':
+            stepSize = self.extinction.initialStep
+            strainRate = self.extinction.initialStrainRate
+        else:
+            incFactor = self.extinction.initialFactor
+            strainRate = self.extinction.initialStrainRate
+
+        complete = False
+        hasExtinguished = False
+
+        while not complete:
+
+            if restartPath is not None:
+                if self.extinction.method is 'step':
+                    strainRate += stepSize
+                else:
+                    strainRate *= incFactor
```

Comment:
I find the logic based on the `restartPath` variable a bit confusing -- I think this could use a few explanatory comments